XEP-0369 MIX-CORE update to sort JID addressing

stevekille opened the pull request.

Kev commented on xep-0369.xml:
```diff
@@ -580,28 +588,31 @@
       </p>
       <p>
-       A Nick MAY be associated with a participant.  The nick associated with the user is stored in a &lt;nick/&gt; child element of the &lt;participant/&gt; element.   The nick for each channel participant MUST be different to the nick of other participants.
+       A Nick MAY be associated with a participant, which provides a user-oriented  description of the participant.  The nick associated with the user is stored in a &lt;nick/&gt; child element of the &lt;participant/&gt; element.   The nick for each channel participant MUST be different to the nick of other participants.
```
I'm not convinced Nicks should be optional (I thought we'd come to that conclusion on-list).

Kev replied:
<Steve> I can see sensible applications for MIX, for example IoT, where Nicks are not sensible.

Kev replied:
So make it a "MUST except..."? Them being optional seems very unhelpful for the majority of cases.

Kev commented on xep-0369.xml:
```diff
@@ -580,28 +588,31 @@
+       A Nick MAY be associated with a participant, which provides a user-oriented  description of the participant.  The nick associated with the user is stored in a &lt;nick/&gt; child element of the &lt;participant/&gt; element.   The nick for each channel participant MUST be different to the nick of other participants.
+      </p>
+      <p>
+        Where a nick is provided for a user, it is generally recommended to use this nick or the JID to display the user.   This enables consistent representation of participants for all participants in the channel.
```
Surely we should only recommend showing the nick? JIDs should be hidden from users whenever possible, and in this case a client showing JIDs might have unpleasant interactions in anonymous rooms.

Kev replied:
&lt;Steve&gt; The JID will be known to all. I can see some situations where you’d prefer to use the definitive name. The key point is to not use local references. I’m fine to delete “or the JID” if you prefer.

Kev replied:
I think that would be better.

Kev commented on xep-0369.xml:
```diff
@@ -580,28 +588,31 @@
       </p>
       <p>
-        When a user joins a channel, the user's bare proxy JID is added to the participants node by the MIX service.   When a user leaves a channel, the user's bare proxy JID is removed from the participants node.  The participants node MUST NOT be directly modified using pubsub.
+        When a user joins a channel, an item representing the is added to the participants node by the MIX service.   When a user leaves a channel, the user's item is removed from the participants node.  The participants node MUST NOT be directly modified using pubsub.
```
'an item representing the is added'?

Kev replied:
&lt;Steve&gt; Fixed

Kev commented on xep-0369.xml:
```diff
@@ -668,7 +679,7 @@
     <li>IQ.  All of the IQ errors of &rfc6120; MUST be supported.</li>
     <li>Messages.  If a message is received and an error situation is encountered, a message of type error MUST be sent back to the message sender. This message format is specified in &rfc6121; containing an error defined in &rfc6120;, which is the same error set as for IQs.</li>
-    <li>Presence.  Any responses to presence messages MUST follow the rules of &rfc6121;.</li>
+    <li>Presence. If MIX-PRESENCE is supported,  any responses to presence messages MUST follow the rules of &rfc6121;.</li>
```
There's no such thing as a "presence message".
It's not clear to me what following the rules of 6121 would mean in this case?

Kev replied:
&lt;Steve&gt; Corrected

Kev replied:
> I’m not sure what to write here.

What is the text trying to say?

Kev commented on xep-0369.xml:
```diff
@@ -997,7 +1010,7 @@
 </message>
 ]]></example>
-      <p>The user that has been added to the channel is identified by the item id of the item added to the Participants node, which is the proxy JID of the new channel participant.  The &lt;participant&gt; element MUST include a &lt;jid&gt; element with the JID of the participant, unless MIX-ANON is being followed to hide participant JIDs.    The &lt;nick&gt; element will not be included at this  ...
+      <p>The user that has been added to the channel is identified by the item id of the item added to the Participants node, which is the Stable Participant ID of the new channel participant.  The &lt;participant&gt; element MUST include a &lt;jid&gt; element with the JID of the participant, unless MIX-ANON is being followed to hide participant JIDs.    The &lt;nick&gt; element will not be inclu ...
```
I thought the plan was that you specified your nick when joining? Having a situation where participants are nick-less because their join processing didn't complete seems unhelpful.

Kev replied:
&lt;Steve&gt; The protocol has setting Nicks independent of Join (it has always been this way). I don’t recollect any plan to change this.

Kev replied:
0.1 had a note to include the nickname in the join, but the text wasn't yet written.

Kev commented on xep-0369.xml:
```diff
@@ -1132,9 +1145,11 @@
+        <li>A &lt;jid&gt; element containing the real JID of the sender. This MUST be present, unless following the "JID Hidden" model defined in MIX-ANON.  If this element is omitted, the &lt;nick&gt; element MUST be present.</li>
       </ol>
+      <p>
+        MIX messages are distributed by the channel with the from using the JID of the channel, with the Stable Participant ID of the sender in the resource.    This enables a receiving system to distinguish messages based on sender using on the JID.   
```
Using on...ly?

Kev replied:
&lt;Steve&gt; Fixed

stevekille commented:
From: Kevin Smith <notifications@github.com> 
Sent: 05 June 2018 15:55
To: xsf/xeps <xeps@noreply.github.com>
Cc: Steve Kille <steve.kille@isode.com>; Author <author@noreply.github.com>
Subject: Re: [xsf/xeps] XEP-0369 MIX-CORE update to sort JID addressing (#655)

 

@Kev commented on this pull request.

Does this not also need the addressing rules for direct routing?

[Steve Kille]

In MIX-PRESENCE.   Core does not need them.

  _____  

In xep-0369.xml <https://github.com/xsf/xeps/pull/655#discussion_r193100076> :

>        </p>
       <p>
-       A Nick MAY be associated with a participant.  The nick associated with the user is stored in a &lt;nick/&gt; child element of the &lt;participant/&gt; element.   The nick for each channel participant MUST be different to the nick of other participants.
+       A Nick MAY be associated with a participant, which provides a user-oriented  description of the participant.  The nick associated with the user is stored in a &lt;nick/&gt; child element of the &lt;participant/&gt; element.   The nick for each channel participant MUST be different to the nick of other participants.

I'm not convinced Nicks should be optional (I thought we'd come to that conclusion on-list).

[Steve Kille]

I can see sensible applications for MIX, for example IoT, where Nicks are not sensible.

  _____  

In xep-0369.xml <https://github.com/xsf/xeps/pull/655#discussion_r193100445> :

>        </p>
       <p>
-       A Nick MAY be associated with a participant.  The nick associated with the user is stored in a &lt;nick/&gt; child element of the &lt;participant/&gt; element.   The nick for each channel participant MUST be different to the nick of other participants.
+       A Nick MAY be associated with a participant, which provides a user-oriented  description of the participant.  The nick associated with the user is stored in a &lt;nick/&gt; child element of the &lt;participant/&gt; element.   The nick for each channel participant MUST be different to the nick of other participants.
+      </p>
+      <p>
+        Where a nick is provided for a user, it is generally recommended to use this nick or the JID to display the user.   This enables consistent representation of participants for all participants in the channel.

Surely we should only recommend showing the nick? JIDs should be hidden from users whenever possible, and in this case a client showing JIDs might have unpleasant interactions in anonymous rooms.

[Steve Kille]

The JID will be known to all.   I can see some situations where you’d prefer to use the definitive name.  The key point is to not use local references.    I’m fine to delete “or the JID” if you prefer.

 

  _____  

In xep-0369.xml <https://github.com/xsf/xeps/pull/655#discussion_r193100830> :

>        </p>
       <p>
         The real JID of the user MAY be held in the participants node.   When the real JID is not present, the procedures defined in MIX-ANON must be followed.
         The user's JID is stored in a &lt;jid/&gt; child element of the &lt;participant/&gt; element.
       </p>
       <p>
-        When a user joins a channel, the user's bare proxy JID is added to the participants node by the MIX service.   When a user leaves a channel, the user's bare proxy JID is removed from the participants node.  The participants node MUST NOT be directly modified using pubsub.
+        When a user joins a channel, an item representing the is added to the participants node by the MIX service.   When a user leaves a channel, the user's item is removed from the participants node.  The participants node MUST NOT be directly modified using pubsub.

'an item representing the is added'?

[Steve Kille]

Fixed

  _____  

In xep-0369.xml <https://github.com/xsf/xeps/pull/655#discussion_r193101410> :

> @@ -668,7 +679,7 @@
   <ol>
     <li>IQ.  All of the IQ errors of &rfc6120; MUST be supported.</li>
     <li>Messages.  If a message is received and an error situation is encountered, a message of type error MUST be sent back to the message sender. This message format is specified in &rfc6121; containing an error defined in &rfc6120;, which is the same error set as for IQs.</li>
-    <li>Presence.  Any responses to presence messages MUST follow the rules of &rfc6121;.</li>
+    <li>Presence. If MIX-PRESENCE is supported,  any responses to presence messages MUST follow the rules of &rfc6121;.</li>

There's no such thing as a "presence message".

[Steve Kille]

Corrected


It's not clear to me what following the rules of 6121 would mean in this case?

[Steve Kille

I’m not sure what to write here.

  _____  

In xep-0369.xml <https://github.com/xsf/xeps/pull/655#discussion_r193102299> :

> @@ -997,7 +1010,7 @@
   </event>
 </message>
 ]]></example>
-      <p>The user that has been added to the channel is identified by the item id of the item added to the Participants node, which is the proxy JID of the new channel participant.  The &lt;participant&gt; element MUST include a &lt;jid&gt; element with the JID of the participant, unless MIX-ANON is being followed to hide participant JIDs.    The &lt;nick&gt; element will not be included at this point, unless it is automatically generated by the channel.   In the likely event that a Nick is subsequently added, an update with the &lt;nick&gt; element will be distributed.
+      <p>The user that has been added to the channel is identified by the item id of the item added to the Participants node, which is the Stable Participant ID of the new channel participant.  The &lt;participant&gt; element MUST include a &lt;jid&gt; element with the JID of the participant, unless MIX-ANON is being followed to hide participant JIDs.    The &lt;nick&gt; element will not be included at this point, unless it is automatically generated by the channel.   In the likely event that a Nick is subsequently added, an update with the &lt;nick&gt; element will be distributed.

I thought the plan was that you specified your nick when joining? Having a situation where participants are nick-less because their join processing didn't complete seems unhelpful.

[Steve Kille]

The protocol has setting Nicks independent of Join (it has always been this way).   I don’t recollect any plan to change this.   

  _____  

In xep-0369.xml <https://github.com/xsf/xeps/pull/655#discussion_r193103554> :

>        </ol>
+      <p>
+        MIX messages are distributed by the channel with the from using the JID of the channel, with the Stable Participant ID of the sender in the resource.    This enables a receiving system to distinguish messages based on sender using on the JID.   

Using on...ly?

[Steve Kille]
Fixed

 

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub <https://github.com/xsf/xeps/pull/655#pullrequestreview-126004440> , or mute the thread <https://github.com/notifications/unsubscribe-auth/AMbljDxMbDKxmt4YfrORlEZKtZBT9PxEks5t5puxgaJpZM4Ua8nL> .  <https://github.com/notifications/beacon/AMbljG1oMJfbHdiB1GjEIatXWaySgf5mks5t5puxgaJpZM4Ua8nL.gif> 

[Steve Kille]

Steve

stevekille commented:
I’m not sure what you are saying here or what change you want

 


S

 

From: Kevin Smith <notifications@github.com> 
Sent: 05 June 2018 16:32
To: xsf/xeps <xeps@noreply.github.com>
Cc: Steve Kille <steve.kille@isode.com>; Author <author@noreply.github.com>
Subject: Re: [xsf/xeps] XEP-0369 MIX-CORE update to sort JID addressing (#655)

 

@Kev commented on this pull request.

  _____  

In xep-0369.xml <https://github.com/xsf/xeps/pull/655#discussion_r193118563> :

>        </p>
       <p>
-       A Nick MAY be associated with a participant.  The nick associated with the user is stored in a &lt;nick/&gt; child element of the &lt;participant/&gt; element.   The nick for each channel participant MUST be different to the nick of other participants.
+       A Nick MAY be associated with a participant, which provides a user-oriented  description of the participant.  The nick associated with the user is stored in a &lt;nick/&gt; child element of the &lt;participant/&gt; element.   The nick for each channel participant MUST be different to the nick of other participants.

I can see sensible applications for MIX, for example IoT, where Nicks are not sensible.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub <https://github.com/xsf/xeps/pull/655#discussion_r193118563> , or mute the thread <https://github.com/notifications/unsubscribe-auth/AMbljO3bl-D53d5fAsoMEHGdou9eNRKfks5t5qRjgaJpZM4Ua8nL> .  <https://github.com/notifications/beacon/AMbljGaHRmQWTEZJn3QdxevfaWMy2Fhgks5t5qRjgaJpZM4Ua8nL.gif>

stevekille commented:
Just figured it.   These are MY comments under @kev

 

From: Kevin Smith <notifications@github.com> 
Sent: 05 June 2018 16:34
To: xsf/xeps <xeps@noreply.github.com>
Cc: Steve Kille <steve.kille@isode.com>; Author <author@noreply.github.com>
Subject: Re: [xsf/xeps] XEP-0369 MIX-CORE update to sort JID addressing (#655)

 

@Kev commented on this pull request.

  _____  

In xep-0369.xml <https://github.com/xsf/xeps/pull/655#discussion_r193119316> :

>        </p>
       <p>
-       A Nick MAY be associated with a participant.  The nick associated with the user is stored in a &lt;nick/&gt; child element of the &lt;participant/&gt; element.   The nick for each channel participant MUST be different to the nick of other participants.
+       A Nick MAY be associated with a participant, which provides a user-oriented  description of the participant.  The nick associated with the user is stored in a &lt;nick/&gt; child element of the &lt;participant/&gt; element.   The nick for each channel participant MUST be different to the nick of other participants.
+      </p>
+      <p>
+        Where a nick is provided for a user, it is generally recommended to use this nick or the JID to display the user.   This enables consistent representation of participants for all participants in the channel.

<Steve> The JID will be known to all. I can see some situations where you’d prefer to use the definitive name. The key point is to not use local references. I’m fine to delete “or the JID” if you prefer.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub <https://github.com/xsf/xeps/pull/655#discussion_r193119316> , or mute the thread <https://github.com/notifications/unsubscribe-auth/AMbljDyQOY1riUsmWhLhnEK5MdJefaLVks5t5qThgaJpZM4Ua8nL> .  <https://github.com/notifications/beacon/AMbljIuqG426H-ftCnShAuv3MxM-7-ykks5t5qThgaJpZM4Ua8nL.gif>

Kev commented:
> In MIX-PRESENCE. Core does not need them.

I don't believe that's true.
